cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java

```diff
@@ -136,6 +136,7 @@ public void create(HttpRequest request, final HttpResponder responder,
     // Enable ad-hoc exploration of dataset
     // Note: today explore enable is not transactional with dataset create - CDAP-8
     try {
+      datasetExploreFacade.disableExplore(name);
```

Comment:
Good point. It works, but it logs an error. I'll update as to not run the delete if table is not there.